### PR TITLE
Localization for numbers throughout the ResearchKit

### PR DESCRIPTION
**Pull request by umerkhan-apple**
The following have been updated and localized:
 - Step progress number for active tasks (e.g. Step 1 of 5)
 - Hz number in Tone Audiometry Active Task
 - Attempt count in Reaction Time Active Task
 - Count down timer for active tasks

### CODE REVIEW

**Comment by julientherier** [comment]
I've seen that the tapping count is not localized too.
Is it possible to localize the numeric keyboard for a numeric question step?

File: ResearchKit/ActiveTasks/ORKCountdownStepViewController.m [hunk 1]
```diff
@@ -229,7 +229,7 @@ - (void)viewDidAppear:(BOOL)animated {
 }
 
 - (void)updateCountdownLabel {
-    _countdownView.timeLabel.text = [NSString stringWithFormat:@"%lu", (unsigned long)_countDown];
+    _countdownView.timeLabel.text = [NSNumberFormatter localizedStringFromNumber:[NSNumber numberWithInteger:_countDown] numberStyle:NSNumberFormatterNoStyle];
```
Comment (YuanZhu-apple) [inline ResearchKit/ActiveTasks/ORKCountdownStepViewController.m]: Prefer the new syntax: `@(_countDown)`

File: ResearchKit/ActiveTasks/ORKToneAudiometryStepViewController.m [hunk 2]
```diff
@@ -200,9 +200,13 @@ - (void)startCurrentTest {
     NSNumber *frequency = self.testingFrequencies[frequencyIndex];
     ORKAudioChannel channel = ((testIndex % 2) == 0) ? ORKAudioChannelLeft : ORKAudioChannelRight;
 
+    NSString *frequencyLocalizedString = [NSString stringWithFormat:@"%@", [NSNumberFormatter
+                                                       localizedStringFromNumber:frequency
+                                                       numberStyle:NSNumberFormatterNoStyle]];
+
```
Comment (YuanZhu-apple) [inline ResearchKit/ActiveTasks/ORKToneAudiometryStepViewController.m]: What is the purpose of `stringWithFormat:`?

File: ResearchKit/Common/ORKTaskViewController.m [hunk 3]
```diff
@@ -910,7 +910,13 @@ - (void)showViewController:(ORKStepViewController *)viewController goForward:(BO
         ORKTaskProgress progress = [_task progressOfCurrentStep:viewController.step withResult:[self result]];
 
         if (progress.total > 0) {
-            progressLabel = [NSString stringWithFormat:ORKLocalizedString(@"STEP_PROGRESS_FORMAT", nil) ,(unsigned long)progress.current+1, (unsigned long)progress.total];
+            NSString *currentStepLocalizedString = [NSNumberFormatter
+                                                    localizedStringFromNumber:[NSNumber numberWithUnsignedLong:progress.current+1]
+                                                    numberStyle:NSNumberFormatterNoStyle];
```
Comment (YuanZhu-apple) [inline ResearchKit/Common/ORKTaskViewController.m]: Might be good to create a macro `ORKLocalizedStringFromNumber()` in `ORKDefines_Private.h`, so we can make these code cleaner.

File: ResearchKit/Localized/ca.lproj/Localizable.strings [hunk 4]
```diff
@@ -75,7 +75,7 @@
 "CONSENT_DOC_LINE_DATE" = "Data";
 
 /* Progress indication (in navigation bar). */
-"STEP_PROGRESS_FORMAT" = "Pas %1$lu de %2$lu";
+"STEP_PROGRESS_FORMAT" = "Pas %@ de %@";
```
Comment (YuanZhu-apple) [inline ResearchKit/Localized/ca.lproj/Localizable.strings]: Please keep these special tags, e.g. %**1$**ln.
See here: https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/LoadingResources/Strings/Strings.html

File: ResearchKit/ActiveTasks/ORKTappingContentView.m [hunk 5]
```diff
@@ -116,7 +116,10 @@ - (void)tintColorDidChange {
 }
 
 - (void)setTapCount:(NSUInteger)tapCount {
-    _tapCountLabel.text = [NSString stringWithFormat:@"%02lu", (unsigned long)tapCount];
+    NSNumberFormatter *formatter = [NSNumberFormatter new];
+    formatter.locale = [NSLocale currentLocale];
+    formatter.minimumIntegerDigits = 2;
+    _tapCountLabel.text = [NSString stringWithFormat:@"%2@", [formatter stringFromNumber:@(tapCount)]];
```
Comment (YuanZhu-apple) [inline ResearchKit/ActiveTasks/ORKTappingContentView.m]: The tapCount is being updated very frequently, better to retain this formatter with `ORKLocalizedStringFromNumber`(create once).

Also `stringWithFormat:` is redundant.

File: ResearchKit/Common/ORKDefines_Private.h [hunk 6]
```diff
@@ -45,3 +45,5 @@ ORK_EXTERN NSDateComponents *ORKTimeOfDayComponentsFromString(NSString *string)
 ORK_EXTERN NSDateFormatter *ORKResultDateTimeFormatter() ORK_AVAILABLE_DECL;
 ORK_EXTERN NSDateFormatter *ORKResultTimeFormatter() ORK_AVAILABLE_DECL;
 ORK_EXTERN NSDateFormatter *ORKResultDateFormatter() ORK_AVAILABLE_DECL;
+
+ORK_EXTERN NSString *ORKLocalizedStringFromNumber(NSNumber *number) ORK_AVAILABLE_DECL;
```
Comment (YuanZhu-apple) [inline ResearchKit/Common/ORKDefines_Private.h]: Can you make it a macro to align with `ORKLocalizedString`?

**Comment by YuanZhu-apple** [comment]
r=me.

**Comment by umerkhan-apple** [comment]
@julientherier I've committed the code to localize the tapping count as well. Unfortunately, it is not possible to localize the system numeric keyboard.

**Comment by YuanZhu-apple** [comment]
Merged.

**Pull request merged**